ecl/hthor/hthorkey.cpp

```diff
@@ -2597,7 +2598,7 @@ class XmlFetchPartHandler : public SimpleFetchPartHandlerBase, public IXMLSelect
                 part->getFilename(rfn).getPath(fname);
                 throw owner.makeWrappedException(e, fname.str());
             }
-            if(!gotNext)
+            if(!gotNext && !lastMatch) //unfortunately json parser next() has slightly different behavior, tricky, may fix later
```

Comment:
It's certainly better to emulate the same behavior.  I'll look at what it would take and what the impact might be to other uses of the interface.